examples/src/binauraliser_nf/binauraliser_nf_internal.h

```diff
@@ -0,0 +1,107 @@
+ * REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
+ * AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
+ * INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
+ * LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
+ * OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
+ * PERFORMANCE OF THIS SOFTWARE.
+ */
+
+/**
+ * @file: binauraliser_nf_internal.h
+ * @brief Convolves input audio (up to 64 channels) with interpolated HRTFs in
+ *        the time-frequency domain, and applies optional near-field binaural
+ *        filtering, as described in [1].
+ *
+ * The HRTFs are interpolated by applying amplitude-preserving VBAP gains to the
+ * HRTF magnitude responses and inter-aural time differences (ITDs)
+ * individually, before being re-combined. The example also allows the user to
+ * specify an external SOFA file for the convolution, and rotations of the
+ * source directions to accomodate head-tracking.
+ *
+ * @see [1] S. Spagnol, E. Tavazzi, and F. Avanzini, “Distance rendering and
+ *          perception of nearby virtual sound sources with a near-field filter
+ *          model,” Applied Acoustics, vol. 115, pp. 61–73, Jan. 2017,
+ *          doi: 10.1016/j.apacoust.2016.08.015.
+ *
+ * @author Michael McCrea, Leo McCormack
+ * @date 22.02.2022
+ * @license ISC
+ */
+
+#ifndef __BINAURALISER_NF_INTERNAL_H_INCLUDED__
+#define __BINAURALISER_NF_INTERNAL_H_INCLUDED__
+
+#include "../binauraliser/binauraliser_internal.h"
+#include <binauraliser_nf.h>    /* Include header for this example */
+#include "saf.h"                /* Main include header for SAF */
+#include "saf_externals.h"      /* To also include SAF dependencies (cblas etc.) */
+
+#ifdef __cplusplus
+extern "C" {
+#endif /* __cplusplus */
+
+/* ========================================================================== */
+/*                            Internal Parameters                             */
+/* ========================================================================== */
+
+/* ========================================================================== */
+/*                             Presets + Constants                            */
+/* ========================================================================== */
+
+/* ========================================================================== */
+/*                                 Structures                                 */
+/* ========================================================================== */
+
+/**
+ * Main structure for binauraliserNF. Contains all variables from binauraliser
+ * (audio buffers, afSTFT, HRTFs, internal variables, flags, user parameters) plus
+ * those specific to the near field variant.
+ */
+typedef struct _binauraliserNF {
+    struct _binauraliser;               /**< "inherit" member vars of binauraliser struct */

```

Comment:
Note anonymous struct requiring extra compile flags `-fms-extensions` and `-Wno-microsoft`.